lib/utils.pm

```diff
@@ -528,9 +529,12 @@ sub is_boot_encrypted {
     return 0 if get_var('UNENCRYPTED_BOOT');
     return 0 if !get_var('ENCRYPT') && !get_var('FULL_LVM_ENCRYPT');
     # for Leap 42.3 and SLE 12 codestream the boot partition is not encrypted
-    # Only aarch64 needs separate handling
+    # Only aarch64 needs separate handling, it has unencrypted boot for fresh
+    # installations, but has encrypted boot if cancel activation of existing
+    # encrypted partitions
     # ppc64le on pre-storage-ng boot was part of encrypted LVM
-    return 0 if !get_var('FULL_LVM_ENCRYPT') && !is_storage_ng && !get_var('OFW');
+    return 0 if !get_var('FULL_LVM_ENCRYPT') && !is_storage_ng && !get_var('OFW')
+      && !(get_var('ENCRYPT_CANCEL_EXISTING') && get_var('ENCRYPT') && is_aarch64());
```

Comment:
i would like to suggest use of encapsulation of long conditions like that. Put them in a subroutine with a appropriate name so they can be reused and be debugged easily in the future. What do you think?

Reply:
This function is already like that. Problem is that it's really corner case. So name would be like 
is_arch_cryptlvm_reinstallation_cancel_activation_of_the_encrypted.
So I doubt that it helps, especially knowing that this condition will still remain.
Other than that it's a hot-patch. We saw that we can have strict expectations and should solve it by scheduling corresponding test modules instead of such conditions. So, my motivation is to fix this to make it work (test fails for 6 months already).

Reply:
are you saying that this is temporal solution? I cant understand what the "We saw that we can have strict expectations" means. 
this is more a suggestion than a request. i think that those long conditions can get really messy and makes the scheduler harder to debug. i can see that to name this is not so obviously defined. another approach would be to break the condition in two part so you can say if is_arch_cryptlvm() && reinstallation_cancel_activation_of_the_encrypted()

Reply:
Again, I'm with you and I believe that this approach is wrong from the beginning. We could simply have two different module, one of which expects encrypted boot and other doesn't. Then it would be clean. I've tried to explain the condition in the comment. I don't think encapsulation will help much, but if you say so, it might be helpful for others too.